src/main.py

```diff
@@ -0,0 +1,371 @@
+                get_reply=self.get_start_reply,
+                on_message=self.on_start_message,
+                on_callback=self.on_start_callback),
+            'join': DispTuple(
+                get_menu=self.get_join_menu,
+                get_reply=self.get_join_reply,
+                on_message=self.on_join_message,
+                on_callback=self.on_join_callback),
+            'session': DispTuple(
+                get_menu=self.get_session_menu,
+                get_reply=self.get_session_reply,
+                on_message=self.on_session_message,
+                on_callback=self.on_session_callback),
+            'create_character': DispTuple(
+                get_menu=self.get_create_character_menu,
+                get_reply=self.get_create_character_reply,
+                on_message=self.on_create_character_message,
+                on_callback=self.on_create_character_callback)
+        }
+
+    async def on_chat_message(self, msg):
+        self.logger.debug(f'{Chatter.on_chat_message.__name__}(msg={msg})')
+
+        content_type, chat_type, chat_id = glance(msg)
+        if msg['from']['is_bot']:
+            await bot.sendMessage(chat_id, "C'mon, you're bot, go away")
+            return
+
+        if content_type != 'text':
+            await bot.sendMessage(chat_id, StaticMessage.by_id(s, 'not_text'))
+
+        self.user = User.by_id(s,
+                               msg['from']['id'],
+                               msg['from']['username'])
+
+        status = self.dispatchers[self.state].on_message(msg['text'])
+        s.commit()
+        reply = self.dispatchers[self.state].get_reply()
+        reply_markup = self.dispatchers[self.state].get_menu()
+        reply = Chatter.join_str(reply=reply, status=status)
+        if len(reply) > 4097:
+            reply = StaticMessage.by_id(s, 'message_too_long')
+        if reply_markup:
+            await self.sender.sendMessage(reply, reply_markup=reply_markup)
+        else:
+            await self.sender.sendMessage(reply)
+
+    async def on_callback_query(self, msg):
+        self.logger.debug(f'{Chatter.on_callback_query.__name__}(msg={msg})')
+
+        query_id, from_id, query_data = glance(msg, flavor='callback_query')
+
+        self.user = User.by_id(s,
+                               msg['from']['id'],
+                               msg['from']['username'])
+
+        status = self.dispatchers[self.state].on_callback(query_data)
+        s.commit()
+        reply = self.dispatchers[self.state].get_reply()
+        reply_markup = self.dispatchers[self.state].get_menu()
+        reply = Chatter.join_str(reply=reply, status=status)
+        if len(reply) > 4097:
+            reply = StaticMessage.by_id(s, 'message_too_long')
+        if reply_markup:
+            await self.sender.sendMessage(reply, reply_markup=reply_markup)
+        else:
+            await self.sender.sendMessage(reply)
+
+    @staticmethod
+    def join_str(reply: str, status: str) -> str:
```

Comment:
Function `join_str` can be rewritten by one string with filtering and joining list (or tuple) items.

For example:

```suggestion
    def join_str(*args) -> str:
        return '\n\n'.join(filter(None, args))
```